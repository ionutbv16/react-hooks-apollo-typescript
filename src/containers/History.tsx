import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import { RouteComponentProps } from 'react-router-dom';
import gql from 'graphql-tag';
import styled from 'styled-components';
import CardComp from '../components/Card';
import { HistoryInfo } from '../types';

export const QUERY_HISTORY_INFO = gql`
  query {
    history @client {
      items    
    }
  }
`

const Wrapper = styled.div`
  padding: 20px;
`;

const ActionBar = styled.div`
  display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
`;

const CardsWrapper = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
`;

const History: React.FC<RouteComponentProps> = () => {
   
 const data:any  = useQuery<HistoryInfo>(QUERY_HISTORY_INFO).data

  return (
    <Wrapper>
      <ActionBar>
         History
      </ActionBar>
      {data && data.history && data.history.items.map( (itemHistory:any, index: number) => 
      (<ActionBar key={index}>
         <div key={index}>{itemHistory.date}</div>
         {itemHistory.game.map ( (item: any) => <CardComp key={item.id} id={item.id} name={item.name} height={item.height} hyperdriverating={item.hyperdriveRating} winner={item.winner} type={item.type} />) }
      </ActionBar>
      ))}
      
    </Wrapper>
  );
};

export default History;
