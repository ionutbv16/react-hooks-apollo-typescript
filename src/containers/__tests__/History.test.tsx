import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing';
import { MemoryRouter } from 'react-router';
import wait from 'waait';
import History  from '../History'; 

describe('Test History Page', () => {
it('should render History without error', async () => {
  await act(async () => {
    const component = renderer.create(
      <MockedProvider addTypename={false}>
        <MemoryRouter>
          <History />
        </MemoryRouter>
      </MockedProvider>,
    );
    await wait(0);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
});
