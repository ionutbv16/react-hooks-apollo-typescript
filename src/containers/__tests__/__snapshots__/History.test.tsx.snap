// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test History Page should render History without error 1`] = `
<div
  className="sc-EHOje iafsfZ"
>
  <div
    className="sc-bZQynM fymwvv"
  >
    History
  </div>
</div>
`;

exports[`should render History without error 1`] = `
<div
  className="sc-EHOje iafsfZ"
>
  <div
    className="sc-bZQynM fymwvv"
  >
    History
  </div>
</div>
`;
